src/modules/hip/kernel/log.hpp

```diff
@@ -0,0 +1,114 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__device__ void log_hip_compute(T *srcPtr, d_float8 *src_f8, d_float8 *dst_f8)
+{
+    if constexpr (std::is_same<T, schar>::value)
+        rpp_hip_math_add8_const(src_f8, src_f8, (float4)128);
```

Comment:
Please check if there is any performance impact of this condition check
You can also have multiple overloaded functions for different datatypes and check the performance
https://github.com/ROCm/rpp/blob/master/src/modules/hip/kernel/brightness.hpp

Reply:
Yes, I tried this experiment as per Rajy's suggestion for color temperature and it performed slightly better than our usual overloaded functions.